persistence-api/src/main/java/io/stargate/db/query/RowsUpdated.java

```diff
@@ -0,0 +1,108 @@
+package io.stargate.db.query;
+
+import com.datastax.oss.driver.shaded.guava.common.base.Preconditions;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Specify which rows are updated by a given DML query.
+ *
+ * <p>There is effectively two implementation of this class, corresponding to the 2 ways rows can be
+ * updated in CQL: either a set of specific rows are impacted ()
+ *
+ * <ul>
+ *   <li>
+ * </ul>
+ */
+public abstract class RowsUpdated {
```

Comment:
`RowsUpdated` is a terrible name and I'm on the lookout for a better suggestion.

`RowsUpdated` is only here for the primary keys. It's not called `PrimaryKeys` because it can be either a set of primary keys, or a range of them, and I was looking for a more generic name. Maybe `PrimaryKeySelection` would be better?

But to answer the question, the non-PK modification are in the `BoundDMLQuery#modifications` list.